app/enterprise.go

```diff
@@ -35,9 +36,9 @@ func RegisterDataRetentionInterface(f func(*App) einterfaces.DataRetentionInterf
 	dataRetentionInterface = f
 }
 
-var elasticsearchInterface func(*App) einterfaces.ElasticsearchInterface
+var elasticsearchInterface func(*App) searchengine.SearchEngineInterface
 
-func RegisterElasticsearchInterface(f func(*App) einterfaces.ElasticsearchInterface) {
+func RegisterElasticsearchInterface(f func(*App) searchengine.SearchEngineInterface) {
```

Comment:
We should also create a v6 JIRA ticket to change this function to remove the Elasticsearch part. But if we are already changing `Elasticsearch` field to `SearchEngine`, may be we can do it in this PR itself.

Reply:
Same thing, this is specificly for elasticsearch, is not a "generic" thing, is design to register the elasticsearch instantiation from the enterprise code.